### ISSUE TITLE
diss argument in fviz_nbclust not compatible with cluster::pam

### ISSUE THREAD

**Issue by SergioHilario** [kassambara/factoextra#90]
Hi,

I have noticed that there are two diss arguments: one in fviz_nbclust and another in cluster::pam. Thus, I cannot simultaneously input a dissimiliarity matrix and at the same time tell cluster::pam that the input for the algorithm is already a dissimilarity matrix.

One solution would be to allow x to be not only a data.frame, because if the object inherits "dist", cluster::pam will recognize it and treat it as a dissimilarity matrix.

So basically it's currently impossible to perform cluster number selection with this function and a dissimilarity matrix.

Hope it helps,
Sergio

**Comment by kassambara** [comment]
Implemented (PR #223). `fviz_nbclust()` now accepts a precomputed dissimilarity (`"dist"`) as `x` for `method = "silhouette"`/`"wss"`, passing it to dissimilarity-capable clusterers (e.g. `cluster::pam`, `factoextra::hcut`) — so you no longer need to reconcile two `diss` arguments. Reprex:

``` r
suppressWarnings(suppressPackageStartupMessages({ library(cluster); library(factoextra) }))
# A precomputed dissimilarity is now a valid input for silhouette / wss:
d <- get_dist(scale(USArrests), method = "euclidean")
fviz_nbclust(d, pam, method = "silhouette")
```

![](https://i.imgur.com/IhcL6tw.png)

<sup>Created on 2026-06-25 with [reprex v2.1.1](https://reprex.tidyverse.org)</sup>

Non-dissimilarity methods (`kmeans`/`clara`) and `method = "gap_stat"` raise a clear error (they need raw data) rather than silently mis-clustering a distance matrix. On `master`; next release.
